doc: Add basic upgrade documentation for Helm Charts

Syphdias opened the pull request:
# Describe what this PR does #
Without this patch the READMEs for the Helm Charts do not provide any documentation on how to upgrade to a newer version. There is at least one known issue when updating to a newer versions that is unavoidable as of writing. There is a workaround for the issue which should be documented in the upgrade section.

This is a problem because currently the only way to find this workaround is to go through closed GitHub issues. These might not be around at the time someone needs this information. Furthermore the issue should be communicated to the operator before it occurs.

This patch adds basic documentation for updating the Helm repository, and upgrading the installed release of the Helm Chart. How values can be set is not part of the documentation. If an operator used custom values, e.g. for the secret, they probably already know how to deal with setting values. However, the docs still remind the reader to take values into account.
Reusing the installed values (`--reuse-values`) has lead to problems in past, which is why it is explicitly discouraged. An example for this would be the value `logLevel` which was changed to `sidecarLogLevel`. Reusing values lead to `.Values.sidecarLogLevel` being empty and the `csi-provisioner` not being started due to invalid value `-v=""`. Comparing new values with set values is encouraged.

The workaround for issue #3397 from GitHub is being addressed in the section Know Issues Upgrading.

Signed-off-by: Christian Kugler <syphdias+git@gmail.com>

## Is there anything that requires special attention ##
~~Do you have any questions?~~
~~Is the change backward compatible?~~

Are there concerns around backward compatibility?
I am not very familiar with Helm 2.x. In this case I think the commands for helm 3.x work just the same on 2.x. Some confirmation on that would be nice.

~~Provide any external context for the change, if any.~~

## Related issues ##
#3397

Madhu-1 commented on charts/ceph-csi-cephfs/README.md:
```diff
@@ -50,6 +50,32 @@ After installation succeeds, you can get a status of Chart
+
+#### Known Issues Upgrading
+- When upgrading to version >=3.7.0, you might encounter an error that the
+  CSIDriver cannot be updated (#3397). This is due to the CSIDriver resource not
```
Provide a link to the issue instead of just the issue number.

Madhu-1 commented on charts/ceph-csi-cephfs/README.md:
````diff
@@ -50,6 +50,32 @@ After installation succeeds, you can get a status of Chart
+  being updatable. To work around this you can delete the CSIDriver object by
+  running:
+  ```bash
+  kubectl delete csidriver rbd.csi.ceph.com
````
cephfs.csi.ceph.com

Madhu-1 reviewed (commented):
Thanks, @Syphdias, for taking this up.

Madhu-1 commented:
Please refer to https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#code-contribution-workflow to fix CI issue.

humblec commented on charts/ceph-csi-rbd/README.md:
```diff
@@ -50,6 +50,32 @@ After installation succeeds, you can get a status of Chart
+
+#### Known Issues Upgrading
+- When upgrading to version >=3.7.0, you might encounter an error that the
+  CSIDriver cannot be updated (#3397). This is due to the CSIDriver resource not
```
CSIDriver can be mentioned as `CSI Driver` or even mention  as `RBD CSI Driver`

Syphdias replied:
I went for `CSIDriver` because this is what the k8s resource is named – or `csidriver.storage.k8s.io` in its longer form. https://kubernetes-csi.github.io/docs/csi-driver-object.html

I'll change it in the first instance but in the next two instances I explicitly mean the k8s resource/object which uses the concatenated spelling. Is this acceptable?

Syphdias committed:
doc: Add basic upgrade documentation for Helm Charts

Without this patch the READMEs for the Helm Charts do not provide any
documentation on how to upgrade to a newer version. There is at least
one known issue when updating to a newer versions that is unavoidable as
of writing. There is a workaround for the issue which should be
documented in the upgrade section.

This is a problem because currently the only way to find this workaround
is to go through closed GitHub issues. These might not be around at the
time someone needs this information. Furthermore the issue should be
communicated to the operator before it occurs.

This patch adds basic documentation for updating the Helm repository,
and upgrading the installed release of the Helm Chart. How values can be
set is not part of the documentation. If an operator used custom values,
e.g. for the secret, they probably already know how to deal with setting
values. However, the docs still remind the reader to take values into
account.
Reusing the installed values (`--reuse-values`) has lead to problems in
past, which is why it is explicitly discouraged. An example for this
would be the value `logLevel` which was changed to `sidecarLogLevel`.
Reusing values lead to `.Values.sidecarLogLevel` being empty and the
`csi-provisioner` not being started due to invalid value `-v=""`.
Comparing new values with set values is encouraged.

The workaround for issue #3397 from GitHub is being addressed in the
section Know Issues Upgrading.

Signed-off-by: Christian Kugler <syphdias+git@gmail.com>

Madhu-1 commented on charts/ceph-csi-cephfs/README.md:
```diff
@@ -50,6 +50,34 @@ After installation succeeds, you can get a status of Chart
+#### Known Issues Upgrading
+
+- When upgrading to version >=3.7.0, you might encounter an error that the
+  CephFS CSI Driver cannot be updated (https://github.com/ceph/ceph-csi/issues/3397).
```
```suggestion
CephFS CSI Driver cannot be updated. Please refer to [issue](https://github.com/ceph/ceph-csi/issues/3397) for more details.
```

Syphdias replied:
I also used a bare URL, which I need to fix.

Syphdias replied:
I assume the dedent was an accident?

Madhu-1 replied:
Yes correct , suggestion is to use link instead of just issue URL

Syphdias replied:
Rerunning `make containerized-test TARGET=lint-extras` before pushing this time… – hopefully for the last time :smile:

Syphdias replied:
`2 chart(s) linted, 0 chart(s) failed` :tada:

Madhu-1 replied:
Thanks for testing it locally :100:

Syphdias replied:
I could have avoided half the comments, if I did in the first place

Madhu-1 reviewed (commented):
small nit, change look good to me.

nixpanic commented:
@mergifyio queue

nixpanic commented:
@mergifyio unqueue

This somehow has the `DNM` label?
